engine/src/core/platform/graphics/rendering/rendering_device/buffers/opengl_vertex_array.cpp

```diff
@@ -0,0 +1,49 @@
+//
+// Created by blinku on 22.07.19.
+//
+
+#include <algorithm>
+#include "opengl_vertex_array.h"
+#include "opengl_buffer.h"
+
+#include "glad/glad.h"
+
+namespace Bald::Platform::Graphics {
+
+    OpenGLVertexArray::OpenGLVertexArray() noexcept : m_ArrayID(0) {
+        glGenVertexArrays(1, &m_ArrayID);
+    }
+
+    OpenGLVertexArray::~OpenGLVertexArray() {
+
+        std::for_each(m_Buffers.begin(), m_Buffers.end(), [](Bald::Graphics::Buffer* buffer) { delete buffer; });
```

Comment:
I think range for loop is better here because it's more natural, readable and then you don't have to include whole algorithm file.